dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/internal/user/DatadogUserInfoProviderTest.kt

```diff
@@ -90,6 +90,49 @@ internal class DatadogUserInfoProviderTest {
         ).isEqualTo(customProperties)
     }
 
+    @Test
+    fun `M use immutable properties W setExtraProperties() is called { changing properties values }`(
```

Comment:
> [!WARNING]
> Why are you testing the `setUserInfo()` on the `DatadogCore` class, but the  `addUserProperties()` on the `DatadogUserInfoProvider` class? Both methods exist on both classes, and it might be relevant to test the same behavior on both classes.

Reply:
because the immutability is handled on different places. The `addUserProperties` is handled inside the `DatadogUserInfoProvider` and that's a mock in the `DatadogCoreTesting`. There are 2 different logics that we are testing here.

Reply:
IMO We need to test both. The DatadogCore should make sure that we don't transfer the immutability as it doesn't know how the UserInfoProvider works, and same from the DatadogUserInfoProvider point of view

Reply:
I don't agree with handling this in 2 places. Testing this in the DatadogCore is not possible as the DatadogCore only delegates to `DatadodUserInfoProvider` which already creates a copy from `original` + `new properties`. If we choose to handle this here also it will create an extra copy. I don't want to exchange tests for less optimised code.

Reply:
In this case let's put all the check in the userInfoProvider. It doesn't make sense to have the checks spread in two locations